src/swarm_algorithms/swarm_algorithm.py

```diff
@@ -1,22 +1,50 @@
+import numpy as np
+
+
 class SwarmIntelligence:
-    def __init__(self, population_size, nb_features, constraints):
+    def __init__(self, population_size, nb_features, constraints, seed=0xCAFFE):
         self.population_size = population_size
         self.nb_features = nb_features
         self.global_best_solution = None
         self.local_best_solutions = None
         self.population = None
         self.constraints = constraints
+        self.fit_function = None
+        self._compiled = False
 
-        self.populate_swarm()
+        self._seed = seed
+        self._rng = np.random.RandomState(seed=self._seed)
 
-    def populate_swarm(self):
-        raise NotImplementedError
+    def compile(self, fit_function, search_space_boundaries):
+        """
+        `Compiles` model so it will use particular fitness function and populates swarm.
+        :param fit_function: Fit function to use
+        :param search_space_boundaries:
+            Search space for each dimension. It should be in form of list of tuples: [(b_{min}, b_{max}), ...],
+            where num of tuples is equal to number of dimensions
+        :return: None
+        """
+        self.fit_function = fit_function
+        self.populate_swarm(search_space_boundaries)
+        self._compiled = True
 
-    def get_new_positions(self, step):
+    def populate_swarm(self, search_space_boundaries):
+        search_space_boundaries = np.array(search_space_boundaries)
+        minimums = search_space_boundaries[:, 0]
+        maxes = search_space_boundaries[:, 1]
+        population = self._rng.uniform(minimums, maxes, size=(self.population_size, self.nb_features))
+        self.population = population
```

Comment:
Population is generated within min-max boundaries, but elements are not checked whether they meet `constraints`